orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -499,9 +499,20 @@ at::Tensor _copy_from_and_resize(
   const auto ort_self = create_ort_value(invoker, self);
   auto ort_dst = create_ort_value(invoker, dst);
 
+  // If the size of the dst is not equal to src, re-allocate the dst tensor to
+  // match the src tensor size.
+  if (self.numel() != dst.numel()) {
+    auto* dst_ort_tensor = ort_dst.GetMutable<onnxruntime::Tensor>();
```

Comment:
REVIEW: can we make this leverage the non-mutable version of the tensor?

Reply:
don't you have to have mutable since you are changing it?

Reply:
We are not actually mutating _this_ tensor (we are swapping it out from the impl holding it, but that does not go through this referance.)